src/dawn/CodeGen/Cuda/MSCodeGen.cpp

```diff
@@ -25,10 +26,11 @@
 namespace dawn {
 namespace codegen {
 namespace cuda {
-MSCodeGen::MSCodeGen(std::stringstream& ss, const std::unique_ptr<iir::MultiStage>& ms,
+MSCodeGen::MSCodeGen(std::stringstream& ss, OptimizerContext* context,
+                     const std::unique_ptr<iir::MultiStage>& ms,
                      const std::shared_ptr<iir::StencilInstantiation>& stencilInstantiation,
                      const CacheProperties& cacheProperties)
-    : ss_(ss), ms_(ms), stencilInstantiation_(stencilInstantiation),
+    : ss_(ss), context_(context), ms_(ms), stencilInstantiation_(stencilInstantiation),
```

Comment:
the MS already holds the `stencilInstantiation`, we can fetch the context from there

Reply:
ok